### ISSUE TITLE
False Error Object of type "None" is not subscriptable when dealing with a csv.DictReader

### ISSUE THREAD

**Issue by AngeloPavone** [microsoft/pyright#3680]
**Describe the bug** 
there is a incorrect error when trying to slice a csv.DictReader object

**To Reproduce**
try to slice a csv.DictReader object using objectname.fieldname[1:]

**Expected behavior**
no error as this is valid python3 code and you can see that the object does not have a type of "None"

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/101136391/178091316-f8295ec3-a17b-4ed8-b837-900edd1b2d38.png)
![image](https://user-images.githubusercontent.com/101136391/178091327-55695b33-308b-44d9-a9f3-fe44f08f6c5a.png)

`database_reader = csv.DictReader(table_file)`
`subsequence = database_reader.fieldnames[1:]`

**additional context**
I am just slicing the "name" column from this csv nothing crazy. The code runs without errors

**Comment by erictraut** [comment]
Please provide a self-contained minimal code sample in text form that reproduces the error. I'd prefer not to type in pages of code from a screen shot.

**Comment by erictraut** [comment]
Here's a minimal self-contained code snippet:

```python
import csv

file = open("", 'r')
reader = csv.DictReader(file)
y = reader.fieldnames[1:]
```

The type definition for `fieldnames` comes from the typeshed stub `csv.pyi`, and it is typed as `Sequence[_T] | None`. Here is [the link](https://github.com/python/typeshed/blob/ee09a67c5cf61effaae1bdb894571eac4fc9ac6b/stdlib/csv.pyi#L83).

If you think this type is incorrect, please file a bug in the typeshed issue tracker.

Pyright is issuing the correct type error for the type information it is provided, so this isn't a bug in pyright.
